crates/uv/src/commands/tool/run.rs

```diff
@@ -104,6 +105,37 @@ pub(crate) async fn run(
         return Err(anyhow::anyhow!("Tool command could not be parsed as UTF-8 string. Use `--from` to specify the package name."));
     };
 
+    if let Some(ref f) = from {
+        if Path::new(&f)
+            .extension()
+            .is_some_and(|ext| ext.eq_ignore_ascii_case("py") || ext.eq_ignore_ascii_case("pyw"))
+        {
+            return Err(anyhow::anyhow!(
+                "It looks you have passed a script instead of a package into `--from`"
```

Comment:
Why not use the same suggestion as below? (but with `--from <package>`)

Reply:
We can't suggest `uv run ...` but we can suggest `uvx --from script-py ...` instead